sdk/formrecognizer/ai-form-recognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,125 @@
+# Troubleshoot Azure Form Recognizer client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Form Recognizer client library for Javascript.
+
+## Table of Contents
+
+- [Troubleshooting Errors](#troubleshooting-errors)
+  - [Handling RestError](#handling-resterror)
+  - [Build model error](#build-model-error)
+    - [Invalid training dataset](#invalid-training-data-set)
+    - [Invalid SAS URL](#invalid-sas-url)
+  - [Generic Error](#generic-error)
+- [Unexpected time to build a custom model](#unexpected-time-to-build-a-custom-model)
+- [Form Recognizer errors](#form-recognizer-errors)
+- [Logging](#logging)
+
+## Troubleshooting Errors
+
+### Handling RestError
+
+Form Recognizer service methods throw [`RestError`] on failure.
+
+The RestError raised by the Azure Form Recognizer client library includes detailed error response information that provides useful insights into what went wrong and includes corrective actions to fix common issues.
+This error information can be found in the `RestError#message` property.
+
+### Build model error
+
+Build model errors usually occur when trying to build a custom model. The most common scenarios when this might occur is, if you are building the model with an
+[Invalid data set](#invalid-training-data-set) or an [Invalid SAS Url](#invalid-sas-url).
+
+#### Invalid training data set
+
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set, [here](https://aka.ms/customModelV3).
+
+Example error below:
+
+```bash
+RestError: Invalid argument.
+  {
+    "name": "RestError",
+    "code": "InvalidArgument",
+    "statusCode": 400,
+    .
+    .
+    .
+    "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidParameter",
+         "message": "The parameter ContainerUrl is invalid: Url is null or whitespace"
+       }
+     }
+    }
+  }
```

Comment:
This doesn't say "invalid data set" but "invalid _containerUrl_". Are there cases where the container URL can be fine, but the data in the container are broken?

Reply:
Thanks for catching it, can plan an error accordingly.